#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation 1 0 0 4.71238898038469
  position 0.019875455453401317 2.2603746726155625 0.036752732731437476
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.1445 1.4855
  floorTileSize 0.33 0.33
}
DEF MAZE Group {
  children [
    DEF cornerStone Solid {
      translation -0.7425000000094198 0.03249942611500021 -0.41250000000941994
      rotation 0.7071067661891148 6.417283026477296e-07 -0.7071067961836885 -4.1071894206282114e-10
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(4)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation 0.7424999999905803 0.03249942611500021 -0.41250000000941994
      rotation 0.7071067661891148 6.417283026477296e-07 -0.7071067961836885 -4.1071894206282114e-10
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(5)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation 0.7424999999905803 0.03249942611500021 0.41249999999058
      rotation 0.7071067661891148 6.417283026477296e-07 -0.7071067961836885 -4.1071894206282114e-10
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(6)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF cornerStone Solid {
      translation -0.7425000000094198 0.03249942611500021 0.41249999999058
      rotation 0.7071067661891148 6.417283026477296e-07 -0.7071067961836885 -4.1071894206282114e-10
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(7)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
    }
    DEF hWall Solid {
      translation 3.1872801566677243e-13 0.03244414186001675 -0.41249999981687824
      rotation -0.11642814508290486 0.9931988155962329 0.0007744228781649712 1.1972605513510558e-08
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      name "solid(1)"
      boundingObject USE hWall
      physics Physics {
        density 100
      }
    }
    DEF hWall Solid {
      translation 3.1872801566677243e-13 0.03244414186001675 0.4125000001831217
      rotation -0.11642814508290486 0.9931988155962329 0.0007744228781649712 1.1972605513510558e-08
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      boundingObject USE hWall
      physics Physics {
        density 100
      }
    }
    DEF vWall Solid {
      translation -0.74249999995094 0.03246939280000916 3.6374701650521536e-14
      rotation 0.0016733062732971625 0.9938291462523422 -0.11090909838897937 -6.8250579342465696e-09
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(2)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
    }
    DEF vWall Solid {
      translation 0.7425000000490601 0.03246939280000916 3.6374701650521536e-14
      rotation 0.0016733062732971625 0.9938291462523422 -0.11090909838897937 -6.8250579342465696e-09
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(3)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
    }
    DEF Pole MazeBlock {
      translation -0.9049999999999252 -1.1477699995779411e-06 -0.5774999999999251
      rotation -0.7071066975301022 3.5122709068564255e-07 0.7071068648428956 8.88984616622212e-10
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999252 -1.1477699995710022e-06 -0.24749999999992484
      rotation -0.7071067812170833 4.6574349287342136e-08 0.7071067811560103 8.88984780275382e-10
      name "maze block(20)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.41249999999992476 -1.1477699995710022e-06 -0.247499999999925
      rotation -0.707106689477574 -2.28834220905268e-07 0.707106872895472 8.889851034517746e-10
      name "maze block(21)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992486 -1.1477699995710022e-06 -0.24749999999992478
      rotation 0.7071068764417333 6.67665366662869e-07 -0.7071066859310338 -8.88984750690955e-10
      name "maze block(22)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992498 -1.1477699995710022e-06 -0.2474999999999249
      rotation -0.7071070468614732 -1.1285204265195543e-07 0.7071065155115133 8.889851357881483e-10
      name "maze block(23)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007516 -1.1477699995710022e-06 -0.2474999999999249
      rotation -0.7071068225955308 -1.4151993740676234e-07 0.7071067397775475 8.889849631611929e-10
      name "maze block(25)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.4125000000000752 -1.1477699995710022e-06 -0.247499999999925
      rotation -0.707106689477574 -2.28834220905268e-07 0.707106872895472 8.889851034517746e-10
      name "maze block(26)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999252 -1.1477699995710022e-06 -0.08249999999992502
      rotation -0.707106624114794 -5.555291919829515e-08 0.707106938258264 8.889848150328294e-10
      name "maze block(30)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999248 -1.1477699995779411e-06 -0.08249999999992494
      rotation -0.7071065886243615 3.1401682363961546e-07 0.7071069737486112 8.889848706731455e-10
      name "maze block(31)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992492 -1.1477699995779411e-06 -0.08249999999992495
      rotation 0.7071069233417818 -6.748592813436812e-08 -0.7071066390312813 -8.889848117966896e-10
      name "maze block(32)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992494 -1.1477699995779411e-06 -0.08249999999992494
      rotation 0.7071067762554928 1.426840803086342e-07 -0.7071067861175878 -8.889849345974393e-10
      name "maze block(33)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007507 -1.1477699995779411e-06 -0.08249999999992494
      rotation 0.7071067762554928 1.426840803086342e-07 -0.7071067861175878 -8.889849345974393e-10
      name "maze block(34)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007524 -1.1477699995779411e-06 -0.082499999999925
      rotation -0.7071066282595253 -2.2151127501117257e-07 0.7071069341135021 8.889856260826991e-10
      name "maze block(35)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007514 -1.1477699995779411e-06 -0.08249999999992494
      rotation -0.7071065886243615 3.1401682363961546e-07 0.7071069737486112 8.889848706731455e-10
      name "maze block(36)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.5775000000000748 -1.1477699995710022e-06 -0.08249999999992502
      rotation -0.707106624114794 -5.555291919829515e-08 0.707106938258264 8.889848150328294e-10
      name "maze block(37)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999252 -1.1477699995710022e-06 0.08250000000007499
      rotation -0.707106624114794 -5.555291919829515e-08 0.707106938258264 8.889848150328294e-10
      name "maze block(40)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999248 -1.1477699995779411e-06 0.08250000000007507
      rotation -0.7071065886243615 3.1401682363961546e-07 0.7071069737486112 8.889848706731455e-10
      name "maze block(41)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992492 -1.1477699995779411e-06 0.08250000000007505
      rotation 0.7071069233417818 -6.748592813436812e-08 -0.7071066390312813 -8.889848117966896e-10
      name "maze block(42)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992494 -1.1477699995779411e-06 0.08250000000007507
      rotation 0.7071067762554928 1.426840803086342e-07 -0.7071067861175878 -8.889849345974393e-10
      name "maze block(43)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007507 -1.1477699995779411e-06 0.08250000000007507
      rotation 0.7071067762554928 1.426840803086342e-07 -0.7071067861175878 -8.889849345974393e-10
      name "maze block(44)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007524 -1.1477699995779411e-06 0.08250000000007501
      rotation -0.7071066282595253 -2.2151127501117257e-07 0.7071069341135021 8.889856260826991e-10
      name "maze block(45)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007514 -1.1477699995779411e-06 0.08250000000007507
      rotation -0.7071065886243615 3.1401682363961546e-07 0.7071069737486112 8.889848706731455e-10
      name "maze block(46)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999251 -1.1477699995710022e-06 0.2475000000000752
      rotation 0.7071070187376789 -1.3467695374397966e-07 -0.7071065436353235 -8.889847983092306e-10
      name "maze block(50)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.41249999999992465 -1.1477699995779411e-06 0.24750000000007513
      rotation -0.7071066713847464 -1.250753637072835e-08 0.7071068909883315 8.889848998815435e-10
      name "maze block(51)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992492 -1.1477699995779411e-06 0.24750000000007513
      rotation -0.7071069593244887 -7.966162323916711e-08 0.7071066030485572 8.88984827478076e-10
      name "maze block(52)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992494 -1.1477699995779411e-06 0.2475000000000752
      rotation -0.7071067735320339 3.7977930490725244e-08 0.7071067888410602 8.889849368730211e-10
      name "maze block(53)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007507 -1.1477699995779411e-06 0.2475000000000752
      rotation -0.7071067735320339 3.7977930490725244e-08 0.7071067888410602 8.889849368730211e-10
      name "maze block(54)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007513 -1.1477699995710022e-06 0.2475000000000751
      rotation -0.7071068730440866 1.8542521421594066e-07 0.707106689328972 8.889849541222273e-10
      name "maze block(55)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.4125000000000753 -1.1477699995779411e-06 0.24750000000007513
      rotation -0.7071066713847464 -1.250753637072835e-08 0.7071068909883315 8.889848998815435e-10
      name "maze block(56)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.577500000000075 -1.1477699995710022e-06 0.2475000000000752
      rotation 0.7071070187376789 -1.3467695374397966e-07 -0.7071065436353235 -8.889847983092306e-10
      name "maze block(57)"
      length 0.015
    }
    DEF hBlock MazeBlock {
      translation -0.6599999999996644 -1.0712519995993708e-05 -0.5774999999905018
      rotation -0.06253680794041941 -0.9980205702999054 0.006639948107836019 -4.807121276957113e-09
      name "maze block(10)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.3300000000003356 -1.0712519995993708e-05 0.08250000000949821
      rotation -0.06253696474435268 -0.9980205606508462 0.006639921590976085 -4.807121320421049e-09
      name "maze block(62)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.16500000000033543 -1.0712519995993708e-05 -0.08249999999050175
      rotation -0.0625368761105848 -0.9980205661359963 0.006639931920961966 -4.8071212681584246e-09
      name "maze block(61)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.16500000000033543 -1.0712519995993708e-05 0.08250000000949825
      rotation -0.0625368761105848 -0.9980205661359963 0.006639931920961966 -4.8071212681584246e-09
      name "maze block(60)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.33000000000033547 -1.0712519996000647e-05 -0.2474999999905018
      rotation -0.06253683693071374 -0.9980205687837961 0.006639902948795295 -4.80712124774431e-09
      name "maze block(59)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.4950000000003354 -1.0712519995993708e-05 -0.08249999999050175
      rotation -0.06253690702900297 -0.9980205639836727 0.006639964228673016 -4.807121287318574e-09
      name "maze block(58)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.4950000000003354 -1.0712519996000647e-05 0.2475000000094982
      rotation -0.06253698857902058 -0.998020559123119 0.006639926734862853 -4.807121322119035e-09
      name "maze block(39)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.1650000000003354 -1.0712519995993708e-05 0.2475000000094982
      rotation -0.06253688909682831 -0.9980205651980045 0.006639950598201975 -4.8071212999162755e-09
      name "maze block(29)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.9049999999905014 -1.0712519995993708e-05 -0.33000000000033536
      rotation -9.011429177181979e-14 1 1.0909928421132766e-13 1.5707996986971802
      name "maze block(78)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.41249999999050196 -1.0712519995993708e-05 -3.354579775465132e-13
      rotation -8.996377570305799e-14 1 1.0897764218437662e-13 1.5707996986971802
      name "maze block(65)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.24750000000949837 -1.0712519995993708e-05 0.16499999999966444
      rotation -9.016193883224796e-14 1 1.0909352493941692e-13 1.5707996986971806
      name "maze block(64)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.24749999999050162 -1.0712519995993708e-05 -3.3547069258765153e-13
      rotation -9.020514795371167e-14 1 1.0916154274137226e-13 1.5707996986971806
      name "maze block(63)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.24750000000949834 -1.0712519996000647e-05 -0.16500000000033563
      rotation -9.022088523743635e-14 1 1.0929187073515272e-13 1.5707996986971806
      name "maze block(49)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125000000094981 -1.0712519996000647e-05 -0.16500000000033563
      rotation -9.025184502931744e-14 1 1.0912201677397424e-13 1.5707996986971802
      name "maze block(11)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125000000094985 -1.0712519995993708e-05 0.3299999999996647
      rotation -9.010360152632688e-14 1 1.0912540276307089e-13 1.5707996986971802
      name "maze block(48)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.4949999999996646 -1.0712519996000647e-05 -0.24749999999050176
      rotation -0.06253684027931171 -0.9980205683190015 0.006639941272121278 -4.807121291026664e-09
      name "maze block(4)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.6599999999996644 -1.0712519996000647e-05 -0.08249999999050181
      rotation -0.06253681364829086 -0.9980205698798131 0.006639957491732665 -4.80712125037751e-09
      name "maze block(6)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 3.3544574482586523e-13 -1.0712519995993708e-05 -0.08249999999050171
      rotation -0.06253697384235973 -0.9980205598316959 0.006639959026116242 -4.807121336936394e-09
      name "maze block(8)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 3.3544574482586523e-13 -1.0712519995993708e-05 0.0825000000094983
      rotation -0.06253697384235973 -0.9980205598316959 0.006639959026116242 -4.807121336936394e-09
      name "maze block(9)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.32999999999966445 -1.0712519995993708e-05 0.08250000000949821
      rotation -0.06253696474435268 -0.9980205606508462 0.006639921590976085 -4.807121320421049e-09
      name "maze block(12)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.3299999999996646 -1.0712519996000647e-05 0.24750000000949818
      rotation -0.06253687037007853 -0.998020566348739 0.006639954010264507 -4.807121327430356e-09
      name "maze block(13)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050173 -1.0712519995993708e-05 0.3299999999996646
      rotation -9.010183610798928e-14 1 1.0911161488251887e-13 1.5707996986971806
      name "maze block(18)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050173 -1.0712519995993708e-05 -0.3300000000003354
      rotation -9.010183610798928e-14 1 1.0911161488251887e-13 1.5707996986971806
      name "maze block(2)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050173 -1.0712519995993708e-05 -3.3546007721706317e-13
      rotation -9.009927041846608e-14 1 1.0903138114565106e-13 1.5707996986971806
      name "maze block(17)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.2474999999905016 -1.0712519996000647e-05 -0.1650000000003356
      rotation -9.012240350442235e-14 1 1.0906609500802143e-13 1.5707996986971802
      name "maze block(15)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5774999999905016 -1.0712519996000647e-05 0.32999999999966473
      rotation -9.025672565101308e-14 1 1.0914378184578209e-13 1.5707996986971806
      name "maze block(1)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5774999999905017 -1.0712519995993708e-05 0.1649999999996643
      rotation -9.014222488952763e-14 1 1.0911897066711794e-13 1.5707996986971806
      name "maze block(3)"
      length 0.14
    }
  ]
}
DEF z5164829 E-puck {
  hidden position_0_0 -6.199807218119858e-08
  hidden position_0_1 6.199824542388647e-08
  hidden rotation_1 1 0 0 5.960464477539063e-08
  hidden rotation_2 -1 0 0 5.960464477539063e-08
  translation -0.6599999998333609 -6.396824215123588e-05 -0.3299999999998194
  rotation 2.9767372004007275e-06 0.9999999999955697 2.0761626409159373e-09 -3.141592642223516
  controller "Main"
  camera_width 320
  camera_height 240
  turretSlot [
    Compass {
    }
    DistanceSensor {
      translation -0.03 0 -0.0100001
      rotation 0 1 0 3.14159
      name "left prox"
    }
    DistanceSensor {
      translation 0.03 0 -0.01
      name "right prox"
    }
    DistanceSensor {
      translation 0 0 -0.04
      rotation 0 -1 0 -1.5707953071795862
      name "front prox"
    }
  ]
}
